### ISSUE TITLE
Empty Environment Variable Secret Values in Deployment Settings

### ISSUE THREAD

**Issue by mikefactorial** [microsoft/powerplatform-build-tools#154]
Hey all,
Wondering if this is a bug or expected behavior. 

What we've noticed is when we generate the deployment settings for the import build task if we leave the value for a secret environment variable blank it throws an exception importing the solution. 

The workaround is to remove the node for the secret EV completely from the deployment settings, but this isn't consistent with other environment variables which you can simply provide a blank value and they will be ignored (i.e. won't throw an error but won't set any value on import).

We've added some logic to explicitly remove these secret EVs with empty values in our pipelines before import, but can't see any reason not to treat a blank / empty value the same way as if the environment variable isn't included at all in the deployment settings json.

e.g. 

I have a solution that only contains a single EV of type secret and this deployment settings json works 

```
{
    "EnvironmentVariables":  [
                             ],
    "ConnectionReferences":  [

                             ]
}

```

But the following throws the exception below

```
{
    "EnvironmentVariables":  [
                                 {
                                     "SchemaName":  "mf_SecretVariable",
                                     "Value":  ""
                                 }
                             ],
    "ConnectionReferences":  [

                             ]
}

``` 

```
<OrganizationServiceFault xmlns:i="[http://www.w3.org/2001/XMLSchema-instance"](http://www.w3.org/2001/XMLSchema-instance%22) xmlns="[http://schemas.microsoft.com/xrm/2011/Contracts">](http://schemas.microsoft.com/xrm/2011/Contracts%22%3E)
  <ActivityId>a07918d9-a0f2-4ed2-a26e-645d47f5bd19</ActivityId>
  <ErrorCode>-2147220970</ErrorCode>
  <ErrorDetails xmlns:d2p1="[http://schemas.datacontract.org/2004/07/System.Collections.Generic">](http://schemas.datacontract.org/2004/07/System.Collections.Generic%22%3E)
    <KeyValuePairOfstringanyType>
      <d2p1:key>ApiExceptionSourceKey</d2p1:key>
      <d2p1:value xmlns:d4p1="[http://www.w3.org/2001/XMLSchema"](http://www.w3.org/2001/XMLSchema%22) i:type="d4p1:string">Plugin/Microsoft.Dynamics.EnvironmentVariables.Plugins.CreateVariableValue</d2p1:value>
    </KeyValuePairOfstringanyType>
    <KeyValuePairOfstringanyType>
      <d2p1:key>ApiOriginalExceptionKey</d2p1:key>
      <d2p1:value xmlns:d4p1="[http://www.w3.org/2001/XMLSchema"](http://www.w3.org/2001/XMLSchema%22) i:type="d4p1:string">Microsoft.Xrm.Sdk.InvalidPluginExecutionException: Null or empty secret reference value found. ---&gt; Microsoft.Xrm.Sdk.InvalidPluginExecutionException: Null or empty secret reference value found.
   at Microsoft.Dynamics.EnvironmentVariables.Plugins.EnvironmentVariableUtil.ValidateSecretVariableValue(Entity definition, Entity variableValue, IPluginExecutionContext context, IServiceProvider serviceProvider)
   at Microsoft.Dynamics.EnvironmentVariables.Plugins.CreateVariableValue.Execute(IServiceProvider serviceProvider)
   at Microsoft.Crm.Extensibility.V5PluginProxyStep.ExecuteInternal(PipelineExecutionContext context)
   at Microsoft.Crm.Extensibility.VersionedPluginProxyStepBase.Execute(PipelineExecutionContext context)
   --- End of inner exception stack trace ---
   at Microsoft.Crm.Extensibility.VersionedPluginProxyStepBase.Execute(PipelineExecutionContext context)
   at Microsoft.Crm.Extensibility.PipelineInstrumentationHelper.Execute(Boolean instrumentationEnabled, String stopwatchName, ExecuteWithInstrumentation action, PipelineExecutionContext context)
   at Microsoft.Crm.Extensibility.Pipeline.&lt;&gt;c__DisplayClass3_0.&lt;RunStep&gt;b__0()</d2p1:value>
    </KeyValuePairOfstringanyType>...
```

**Comment by davidjenni** [comment]
It's partially by design, the error you shows is returned from the SolutionsFX in Dataverse. We're looking into finding a mitigation on the pac CLI side to filter out conflicting deployment settings values

AB#2911654

**Comment by davidjenni** [comment]
After discussing with the Dataverse team, we conclude that this is indeed by design: an EnvVar without value is rejected. We added validation logic to the pac solution import side to reject this early, before attempting the import

**Comment by davidjenni** [comment]
Fixed with 2.0.8; note that due to Azure DevOps's caching, it might take several hours until all build agents run with the latest update

https://github.com/microsoft/powerplatform-build-tools/releases/tag/v2.0.8

CC: @mikefactorial

**Comment by dmcweeney** [comment]
Hi @davidjenni , this change just broke our deployments!
Would it be possible to change the behaviour to log a warning instead of a hard stop. 
A few cases here where we dont need the env vars to be correctly set is sandbox environments where devs are working on web resources.

cc: @mikefactorial
